main.mjs

```diff
@@ -84,6 +87,20 @@ app.get("/metrics", async (_, res) => {
         [, completed_total],
       ] = results.slice(i * offset, (i + 1) * offset);
 
+      const bullQueue = new bullmq.Queue(queue, { connection: redis, prefix: BULL_PREFIX });
+
+      const wait_max_age_sec = await bullQueue
+        .getWaiting()
```

Comment:
Let's assume there are `1000` messages in `waiting`. BullMQ will pull all messages and respective fields to calculate the age. In my view, this may bring some problems if a number of messages is too high. 

Suggestion: `BullMQ` uses `ZSET` for the key `waiting`. Can you calculate the age of messages by subtraction of the first and last?

Reply:
Yeah I was also contemplating that. I was considering how to pull it directly from the queue instead but must be honest that I'm not familiar enough with redis syntax to know exactly how to do this correctly.

Another consideration, the last job in the queue isn't necessarily the oldest. If something gets retried for example. Maybe that's fine, but the metric would be "last job age" instead..

For my use case it's an ok compromise. I'm just looking for a metric for how long it takes jobs to move through the queue. Aka average waiting duration + processing duration.

Reply:
Or I guess there's a zset function for getting the highest score perhaps?